tools/generator-go-sdk/main.go

```diff
@@ -10,73 +10,102 @@ import (
+			return &path, nil
+		},
+		outputSubDirectory: "microsoft-graph",
+		versionMapper: func(in string) (*string, error) {
+			version := ""
+			switch in {
+			case "StableV1":
+				version = "v1.0"
+			case "Beta":
+				version = "beta"
+			}
+			if version == "" {
+				return nil, fmt.Errorf("unsupported API version: %q", in)
+			}
+			return &version, nil
+		},
+	},
 }
 
 func main() {
 	input := GeneratorInput{
 		settings: generator.Settings{},
 	}
 
-	input.settings.UseOldBaseLayerFor(
-		// @tombuildsstuff: New Services should now use the `hashicorp/go-azure-sdk` base layer by default
-		// instead of the base layer from `Azure/go-autorest` - as such this list is for compatibility purposes
-		// with services already used in `terraform-provider-azurerm`. These services will be gradually removed
-		// from this list to ensure they're migrated across to using `hashicorp/go-azure-sdk`s base layer.
-
-		// NOTE: also see the list in ./tools/generator-terraform/generator/definitions/template_service_client.go
-		// for services/resources which are auto-generated
-		"ContainerApps",
-		"ContainerInstance",
-		"CosmosDB",
-		"DataShare",
-		"FrontDoor",
-		"Insights",
-		"Kusto",
-		"Maintenance",
-		"ManagedServices",
-		"RecoveryServicesBackup", // error: generating Service "RecoveryServicesBackup" / Version "2023-04-01" / Resource "Operation": generating methods: templating methods (using hashicorp/go-azure-sdk): templating: building methods: building response struct template: existing model "ValidateOperationResponse" conflicts with the operation response model for "Validate"
-		"Security",
-		"SecurityInsights",
-		"ServiceFabric",
-		"ServiceFabricManagedCluster",
-		"ServiceLinker",
-		"SqlVirtualMachine",
-		"Storage",
-		"StreamAnalytics",
-		"Subscription",
-		"TimeSeriesInsights",
-
-		// Automation @ 2022-08-08 uses the new base layer, so let's invert the older versions for now
-		"Automation@2015-10-31",
-		"Automation@2019-06-01",
-		"Automation@2020-01-13-preview",
-		"Automation@2021-06-22",
-
-		// @tombuildsstuff: there's generated resources associated with these three - please check before removing these
-		"ContainerService",
-		"LoadTestService",
-		"ManagedIdentity",
-
-		// @tombuildsstuff: KeyVault requires that the exact casing retrieved from the API is re-sent back to the API
-		// as such will require custom work in the Provider (potentially a custom unmarshaller from the HTTP Body) to support this
-		"KeyVault",
-	)
-
 	var serviceNames string
 
 	f := flag.NewFlagSet("generator-go-sdk", flag.ExitOnError)
 	f.StringVar(&input.apiServerEndpoint, "data-api", "http://localhost:5000", "-data-api=http://localhost:5000")
-	f.StringVar(&input.outputDirectory, "output-dir", "", "-output-dir=../generated-sdk-dev")
+	f.StringVar(&input.outputDirectoryPath, "output-dir", "", "-output-dir=../generated-sdk-dev")
+	f.StringVar(&input.sdkName, "sdk", "resource-manager", "-sdk=resource-manager")
```

Comment:
fwiw we should look to update the automated tooling to specify this

https://github.com/hashicorp/pandora/blob/67702bd186a64af9cd6d3975c9a693ce67d8b638/tools/wrapper-automation/run.go#L75-L81

and the scripts: https://github.com/hashicorp/pandora/tree/main/scripts

probably worth adding a resource manager command to the terraform generator at the same time?

https://github.com/hashicorp/pandora/blob/67702bd186a64af9cd6d3975c9a693ce67d8b638/tools/wrapper-automation/run.go#L95-L102